src/Components/Components/src/ParameterView.cs

```diff
@@ -116,6 +116,17 @@ public TValue GetValueOrDefault<TValue>(string parameterName, TValue defaultValu
             return result;
         }
 
+        internal ParameterView Clone()
+        {
+            if (ReferenceEquals(_frames, _emptyFrames))
+            {
+                return Empty;
+            }
+
+            var dictionary = ToDictionary();
+            return FromDictionary((IDictionary<string, object?>)dictionary);
```

Comment:
Technically a `ParameterView` is ordered, and this form of cloning could lose the ordering. It's almost certainly not going to cause any problems since our APIs for supplying root-component parameters are dictionary-based. Maybe in some future edge case it might matter. If you do want to preserve the ordering, something like the following should work:

```cs
internal ParameterView Clone()
{
    var numEntries = 0;
    foreach (var _ in this)
    {
        numEntries++;
    }

    var cloneBuffer = new RenderTreeFrame[1 + numEntries];
    cloneBuffer[0] = RenderTreeFrame.PlaceholderChildComponentWithSubtreeLength(1 + numEntries);
    Array.Copy(_frames, _ownerIndex + 1, cloneBuffer, 1, numEntries);
    return new ParameterView(ParameterViewLifetime.Unbound, cloneBuffer, 0);
}
```

Caution: I haven't actually run this code to see if it works. Would be good to have a unit test for it if we are going to use it.

Reply:
Alternatively we could start asserting that we don't guarantee the ordering within a `ParameterView`, since I think there are other places already that don't preserve it. It's not something we've documented either way AFAIK.